src/component/component_node.ts

```diff
@@ -18,9 +19,10 @@ export function component(
   name: string | typeof Component,
   props: any,
   key: string,
+  hasToggle: boolean,
```

Comment:
not sure, but why is the toggler not inserted in the compiled template instead?

Reply:
That's what I did in the first commit. I did not like having to set/re-set the key variable in the compiled template. Moreover, in the case we were to have `toggler(key, component(.... key .... ))` the two keys would differ (which is not relevant for the respective algorithms, but still).
This is what we should discuss.